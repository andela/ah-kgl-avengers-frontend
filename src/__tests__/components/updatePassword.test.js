import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import UpdatePassword from '../../components/container/updatePassword';

const mockStore = configureMockStore([thunk]);
const store = mockStore({});

const updateProps = {
  changePassword: jest.fn(),
  match: {},
  history: {},
};
let wrapper;
describe('Testing reset password', () => {
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <UpdatePassword {... updateProps} />
        </Router>
      </Provider>,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should create UpdatePassword snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render UpdatePassword component', () => {
    expect(wrapper.find('UpdatePassword')).toBeDefined();
  });

  it('Should check component props', () => {
    expect(wrapper.find('UpdatePassword').props().changePassword).toBeDefined();
    expect(wrapper.find('UpdatePassword').props().history).toBeDefined();
    expect(wrapper.find('UpdatePassword').props().match).toBeDefined();
  });

  it('Should update password', async () => {
    const button = wrapper.find('button').at(0);
    await button.simulate('click');
    expect(wrapper.state().email).toEqual(undefined);
  });

  it('Should handle input change', () => {
    const input = wrapper.find('Input').at(0);
    input.simulate('change');
    expect(wrapper.find('Input')).toBeDefined();
  });

  it('Should handle form submit event', () => {
    const form = wrapper.find('Form');
    form.simulate('submit');
    expect(form).toBeDefined();
  });
});
