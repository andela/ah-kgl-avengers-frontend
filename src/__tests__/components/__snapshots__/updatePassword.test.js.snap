// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing reset password Should create UpdatePassword snapshot 1`] = `ReactWrapper {}`;
